prov/verbs/src/ep_rdm/verbs_av_ep_rdm.c

```diff
@@ -127,20 +127,23 @@ static int fi_ibv_rdm_av_insert(struct fid_av *av_fid, const void *addr,
 		pthread_mutex_lock(&ep->cm_lock);
 	}
 
-	if (av->type == FI_AV_TABLE) {
-		void *p = realloc(av->domain->rdm_cm->conn_table,
-				(av->count + count * sizeof(*av->domain->rdm_cm->conn_table)));
-		if (p) {
-			av->domain->rdm_cm->conn_table = p;
-		} else {
-			ret = -FI_ENOMEM;
-			goto out;
+	if (av->used + count > av->count) {
+		av->count = av->used + count;
```

Comment:
I would defer changing av->count until after the realloc has succeeded.  Otherwise, the count value will be larger than the actual size.